PSModule/AlmLabMgmtClient/SDK/Handler/EventLogHandler.cs

```diff
@@ -0,0 +1,74 @@
+using PSModule.AlmLabMgmtClient.SDK.Interface;
+using PSModule.AlmLabMgmtClient.SDK.Request;
+using PSModule.AlmLabMgmtClient.SDK.Util;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace PSModule.AlmLabMgmtClient.SDK.Handler
+{
+    public class EventLogHandler : HandlerBase
+    {
+        private const string CREATION_TIME = "creation-time";
+        private const string DESCRIPTION = "description";
+        private const string ID = "id";
+
+        private int _lastRead = -1;
+        private new readonly string _timeslotId;
+
+        public EventLogHandler(IClient client, string timeslotId) : base(client, timeslotId)
+        {
+            _timeslotId = timeslotId;
+        }
+
+        public async Task<bool> Log(bool logRequestUrl)
+        {
+            bool ok = false;
+            Response eventLog = null;
+            try
+            {
+                eventLog = await GetEventLog(logRequestUrl);
+                string xml = eventLog.ToString();
+                var entities = Xml.ToEntities(xml);
+                foreach (var currEntity in entities)
+                {
+                    if (IsNew(currEntity))
+                    {
+                        await _logger.LogInfo($"{currEntity[CREATION_TIME]}:{currEntity[DESCRIPTION]}");
+                    }
+                }
+                ok = true;
+            }
+            catch(ThreadInterruptedException)
+            {
+                throw;
+            }
+            catch (Exception ex)
+            {
+                await _logger.LogError($"Failed to print Event Log: {eventLog} (run id: {_runId}, reservation id: {_timeslotId}). Cause: {ex.Message}");
+
+            }
+
+            return ok;
+        }
+
+        private bool IsNew(IDictionary<string, string> currEntity)
+        {
+            bool isNew = false;
+            int currEvent = int.Parse(currEntity[ID]);

```

Comment:
1. The `ID` may not exist in dictionary.
2. `int.Parse` may throw exception, use `int.TryParse` instead.